themes-default/slim/src/components/subtitle-search.vue

```diff
@@ -0,0 +1,258 @@
+    },
+    destroyed() {
+        // Remove the element from the DOM
+        this.$el.parentNode.removeChild(this.$el);
+    },
+    methods: {
+        autoSearch() {
+            const { episode, season, show } = this;
+
+            this.displayQuestion = false;
+            const url = `home/searchEpisodeSubtitles?indexername=${show.indexer}&seriesid=${show.id[show.indexer]}&season=${season}&episode=${episode}`;
+            this.loadingMessage = 'Searching for subtitles and downloading if available... ';
+            this.loading = true;
+            apiRoute(url) // eslint-disable-line no-undef
+                .then(response => {
+                    if (response.data.result !== 'failure') {
+                        // Update the show, as we have new information (subtitles)
+                        // Let's emit an event, telling the displayShow component, to update the show using the api/store.
+                        this.$emit('update', {
+                            reason: 'new subtitles found',
+                            codes: response.data.subtitles,
+                            languages: response.data.languages
+                        });
+                    }
+                })
+                .catch(error => {
+                    console.log(`Error trying to search for subtitles. Error: ${error}`);
+                })
+                .finally(() => {
+                    // Destroy this component.
+                    this.loadingMessage = '';
+                    this.loading = false;
+                    this.$destroy();
+                });
+        },
+        manualSearch() {
+            const { show, season, episode } = this;
+
+            this.displayQuestion = false;
+            this.loading = true;
+            this.loadingMessage = 'Searching for subtitles... ';
+            const url = `home/manualSearchSubtitles?indexername=${show.indexer}&seriesid=${show.id[show.indexer]}&season=${season}&episode=${episode}`;
+            apiRoute(url) // eslint-disable-line no-undef
+                .then(response => {
+                    if (response.data.result === 'success') {
+                        this.subtitles.push(...response.data.subtitles);
+                    }
+                }).catch(error => {
+                    console.log(`Error trying to search for subtitles. Error: ${error}`);
+                    this.$destroy();
+                }).finally(() => {
+                    this.loading = false;
+                });
+        },
+        pickSubtitle(subtitleId) {
+            // Download and save this subtitle with the episode.
+            const { show, season, episode } = this;
+
+            this.displayQuestion = false;
+            this.loadingMessage = 'downloading subtitle... ';
+            this.loading = true;
+            const url = `home/manualSearchSubtitles?indexername=${show.indexer}&seriesid=${show.id[show.indexer]}&season=${season}&episode=${episode}&picked_id=${subtitleId}`;
```

Comment:
Same as above, but with a note:
Let's say you follow my suggestion and make a computed property for the URL parameters.
Here you have an extra parameter - `picked_id`, so you extend the computed property's value:
```suggestion
            const params = {
                ...episodeParams, // This is the computed property
                picked_id: subtitleId // eslint-disable-line camelcase
            };
```